frontend/src/App.js

```diff
@@ -75,23 +85,71 @@ class App extends React.Component {
           "plantUUID",
           "plantUUID",
           */
-        ]
+        ],
+        gamePhase: "",
+        currentBid: 0
       },
-      powerPlants: {}
+      powerPlants: {},
+      showGame: false,
+      showPowerPlantMarket: false,
+      highestBidder: ""
     };
+    this.toggleShowPowerPlantModal = this.toggleShowPowerPlantModal.bind(this);
+  }
+
+  toggleShowPowerPlantModal() {
+    this.setState(prevState => ({
+      showPowerPlantMarket: !prevState.showPowerPlantMarket
+    }));
   }
 
   componentDidUpdate() {}
 
   componentDidMount() {
-    this.setState({
-      userProfile: userProfile,
-      gameState: gameState,
-      powerPlants: powerPlants
-    });
+    // first ping the API for the first initial state
+    // FAKE API RETURN HERE (actually, we imported it from fakeData.js)
+    // then set initial state so there's a state
+    let self = this;
+    this.setState(
+      {
+        userProfile: userProfile,
+        gameState: gameStatePre,
+        powerPlants: powerPlants
+      },
+      () => {
+        // after first state is set, setInterval to ping gameState API every 100ms
+        this.interval = setInterval(function() {
```

Comment:
nit: I don't think you need the variable `this.interval`. You don't use the result of the return value from `setInterval` anywhere.